app/src/forms/common/models/tables/externalAPI.js

```diff
@@ -0,0 +1,107 @@
+const { Model } = require('objection');
+const { Timestamps } = require('../mixins');
+const { Regex } = require('../../constants');
+const stamps = require('../jsonSchema').stamps;
+
+const { encryptionService } = require('../../../../components/encryptionService');
+
+class ExternalAPI extends Timestamps(Model) {
+  static get tableName() {
+    return 'external_api';
+  }
+
+  static get relationMappings() {
+    const ExternalAPIStatusCode = require('./externalAPIStatusCode');
+
+    return {
+      statusCode: {
+        relation: Model.HasOneRelation,
+        modelClass: ExternalAPIStatusCode,
+        join: {
+          from: 'external_api.code',
+          to: 'external_api_status_code.code',
+        },
+      },
+    };
+  }
+
+  static get modifiers() {
+    return {
+      filterFormId(query, value) {
+        if (value) {
+          query.where('formId', value);
+        }
+      },
+      findByIdAndFormId(query, id, formId) {
+        if (id !== undefined && formId !== undefined) {
+          query.where('id', id).where('formId', formId);
+        }
+      },
+      findByFormIdAndName(query, formId, name) {
+        if (name !== undefined && formId !== undefined) {
+          query.where('name', name).where('formId', formId);
+        }
+      },
+    };
+  }
+
+  async $beforeInsert(context) {
```

Comment:
I'm concerned about encrypting data and then storing it in the database. While this does add some security, such as if someone was able to steal one of the backups, it only adds security if the person is unable to steal the secret key from OpenShift - but the person would still have all the other data in the database. I think that if we have concerns about this sort of attack then we should be looking at encrypting the entire database, and at a much wider scale.

However, if we decide to do this database encryption, then we need to have an easy way of rotating the key, for example: on a regular basis to mitigate problems with the use of AES256GCM using random IVs, after a contractor leaves, or if it accidentally gets exposed.